include/swift/AST/Attr.def

```diff
@@ -699,6 +699,12 @@ CONTEXTUAL_SIMPLE_DECL_ATTR(_const, CompileTimeConst,
   ABIStableToAdd | ABIStableToRemove | APIBreakingToAdd | APIStableToRemove,
   126)
 
+SIMPLE_DECL_ATTR(_unavailableFromAsync, UnavailableFromAsync,
+  OnFunc | OnConstructor |
```

Comment:
I recommend marking this as `UserInaccessible`; that will prevent it from showing up in code completion until we're ready to make it a real feature.

Reply:
Done